evm_arithmetization/src/cpu/kernel/tests/transaction_parsing/parse_type_3_txn.rs

```diff
@@ -0,0 +1,122 @@
+    // encode(signed_tx).hex()[4::] # We don't need the initial RLP prefix.
+    interpreter.extend_memory_segment_bytes(Segment::RlpRaw, hex!("03f8b1820539018203e88201f482520894000000000000000000000000000000000000000201824242c08203e8f842a00101010101010101010101010101010101010101010101010101010101010101a0010101010101010101010101010101010101010101010101010101010101010180a076e2a81a28e69fb1e96f5e9470b454b80663197b416c3783be98a6b5bd162b21a01a182d7a386f81bcbdcc714b2b481b78547b26eaa6b4de417ecd3c1cd53ab839").to_vec());
+
+    interpreter.run()?;
+
+    assert_eq!(interpreter.get_txn_field(ChainIdPresent), 1.into());
+    assert_eq!(interpreter.get_txn_field(ChainId), 1337.into());
+    assert_eq!(interpreter.get_txn_field(Nonce), 1.into());
+    assert_eq!(interpreter.get_txn_field(MaxPriorityFeePerGas), 1000.into());
+    assert_eq!(interpreter.get_txn_field(MaxFeePerGas), 500.into());
+    assert_eq!(interpreter.get_txn_field(To), 0x02.into());
+    assert_eq!(interpreter.get_txn_field(Value), 1.into());
+    assert_eq!(interpreter.get_txn_field(DataLen), 2.into());
+    assert_eq!(interpreter.get_txn_data(), &[0x42.into(), 0x42.into()]);
+    assert_eq!(interpreter.get_txn_field(YParity), 0.into());
+    assert_eq!(
+        interpreter.get_txn_field(R),
+        U256::from_big_endian(&hex!(
+            "76e2a81a28e69fb1e96f5e9470b454b80663197b416c3783be98a6b5bd162b21"
+        ))
+    );
+    assert_eq!(
+        interpreter.get_txn_field(S),
+        U256::from_big_endian(&hex!(
+            "1a182d7a386f81bcbdcc714b2b481b78547b26eaa6b4de417ecd3c1cd53ab839"
+        ))
+    );
+
+    Ok(())
+}
+
+#[test]
+fn process_type_3_txn_invalid_sig() -> Result<()> {
+    let process_type_3_txn = KERNEL.global_labels["process_type_3_txn"];
+    let deduct_blob_fee = KERNEL.global_labels["deduct_blob_fee"];
+
+    let retaddr = 0xDEADBEEFu32.into();
+    const INITIAL_TXN_RLP_ADDR: usize = Segment::RlpRaw as usize + 1;
+    let mut interpreter: Interpreter<F> = Interpreter::new(
+        process_type_3_txn,
+        vec![retaddr, INITIAL_TXN_RLP_ADDR.into()],
+        None,
+    );
+
+    // If we reach deduct_blob_fee, the test fails (we should have had a
+    // kernel panic beforehand).
+    interpreter.halt_offsets.push(deduct_blob_fee);
+
+    // Same transaction as `process_type_3_txn()`, with the exception that the `s`
+    // component in the signature is flipped (i.e. `s' = N - s`, where `N` is the
+    // order of the SECP256k1 prime subgroup).
+    interpreter.extend_memory_segment_bytes(Segment::RlpRaw, hex!("03f8b1820539018203e88201f482520894000000000000000000000000000000000000000201824242c08203e8f842a00101010101010101010101010101010101010101010101010101010101010101a0010101010101010101010101010101010101010101010101010101010101010180a076e2a81a28e69fb1e96f5e9470b454b80663197b416c3783be98a6b5bd162b21a0e5e7d285c7907e4342338eb4d4b7e4866633b5fc0893c1fa4105226ffafb8908").to_vec());
```

Comment:
Here are the diverging parts from a diff:

`..1a182d7a386f81bcbdcc714b2b481b78547b26eaa6[b4d]e417ecd[3c1]cd53ab839`
`..e5e7d285c7907e4342338e[b4d]4b7e4866633b5fc089[3c1]fa4105226ffafb8908`

I notice the two common subarrays of length 3.  Is this to be expected when "flipping" signatures?

Reply:
I think this is rather due to the particular structure of the secp256k1 field and the fact that we're performing a negation (`i.e. s' = N - s`)